trio/tests/test_util.py

```diff
@@ -58,3 +61,54 @@ def handler(signum, _):
                 async with ul1:
                     pass  # pragma: no cover
     assert "ul1" in str(excinfo.value)
+
+
+async def test_contextmanager_do_not_unchain_non_stopiteration_exceptions():
+    @acontextmanager
+    @async_generator
+    async def test_issue29692():
+        try:
+            await yield_()
+        except Exception as exc:
+            raise RuntimeError('issue29692:Chained') from exc
+
+    with pytest.raises(RuntimeError) as excinfo:
+        async with test_issue29692():
+            raise ZeroDivisionError
+    assert excinfo.value.args[0] == 'issue29692:Chained'
+    assert isinstance(excinfo.value.__cause__, ZeroDivisionError)
+
+    # XXX: Broken for StopIteration???
+    for cls in [StopIteration, StopAsyncIteration]:
+        with pytest.raises(cls) as excinfo:
+            async with test_issue29692():
+                raise cls('issue29692:Unchained')
+        assert excinfo.value.args[0] == 'issue29692:Unchained'
+        assert excinfo.value.__cause__ is None
+
+
+# Native async generators are only available from Python 3.6 and onwards
+nativeasyncgenerators = sys.version_info >= (3,6)
+pytestmark = pytest.mark.skipif(not nativeasyncgenerators, reason="Python with native async generators")
```

Comment:
I think this marks the whole file as skipped on 3.5? That doesn't seem like what we want. IIRC the syntax is
```python
# skip a single test
@pytest.mark.skipif(check, reason="whatever")
def test_something():
   ...

# skip multiple tests with the same reason
need_native_async_generators = pytest.mark.skipif(...)

@need_native_async_generators
def test_something():
   ...

@need_native_async_generators
def test_something_else():
   ...
```